akka-docs/src/main/paradox/persistence-fsm.md

```diff
@@ -128,3 +126,87 @@ On recovery state data is initialized according to the latest available snapshot
+
+Java
+:  @@snip [PersistentFsmToTypedMigrationCompileOnlyTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/PersistentFsmToTypedMigrationCompileOnlyTest.java) { #command-handler }
+
+Note that there is no explicit support for state timeout as with PersistentFSM but the same behavior can be achieved
+using `Behaviors.withTimers`. If the timer is the same for all events then it can be hard coded, otherwise the
+old PersistentFSM timeout can be taken from the `StateChangeEvent` in the event adapter and is also available when
+constructing a `SnapshotAdapter`. This can be added to an internal event and then stored in the `State`. Care
+must also be taken to restart timers on recovery in the signal handler:
+
+Scala
+:  @@snip [PersistentFsmToTypedMigrationSpec.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/PersistentFsmToTypedMigrationSpec.scala) { #signal-handler }
+
+Java
+:  @@snip [PersistentFsmToTypedMigrationCompileOnlyTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/PersistentFsmToTypedMigrationCompileOnlyTest.java) { #signal-handler }
+
+Then the event handler:
+
+Scala
+:  @@snip [PersistentFsmToTypedMigrationSpec.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/PersistentFsmToTypedMigrationSpec.scala) { #event-handler }
+
+Java
+:  @@snip [PersistentFsmToTypedMigrationCompileOnlyTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/PersistentFsmToTypedMigrationCompileOnlyTest.java) { #event-handler }
+
+The last step is the adapters that will allow the new @apidoc[EventSourcedBehavior] to read the old data:
+
+Scala
+:  @@snip [PersistentFsmToTypedMigrationSpec.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/PersistentFsmToTypedMigrationSpec.scala) { #event-adapter }
+
+Java
+:  @@snip [PersistentFsmToTypedMigrationCompileOnlyTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/PersistentFsmToTypedMigrationCompileOnlyTest.java) { #event-adapter }
+
+The snapshot adapter needs to adapt an internal type of PersistentFSM so a helper function is provided to build the @apidoc[SnapshotAdapter]:
+
+Scala
+:  @@snip [PersistentFsmToTypedMigrationSpec.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/PersistentFsmToTypedMigrationSpec.scala) { #snapshot-adapter }
+
+Java
+:  @@snip [PersistentFsmToTypedMigrationCompileOnlyTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/PersistentFsmToTypedMigrationCompileOnlyTest.java) { #snapshot-adapter }
+
+That concludes all the steps to allow an @apidoc[EventSourcedBehavior] to read a `PersistentFSM`'s data. Once the new code has been running
+you can not roll back as the PersistentFSM will not be able to read data written by Persistence Typed. This means that if using persistence
```

Comment:
Do we want to add an entry in the migration guide or https://github.com/akka/akka/blob/master/akka-docs/src/main/paradox/additional/rolling-updates.md for this? IDK.

Reply:
I see you did below :-)